packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/blocs/onboarding_bloc.dart

```diff
@@ -0,0 +1,189 @@
+{{> licence.dart }}
+
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_services/users_service.dart';
+import '../../base/common_services/validators/login_validator_service.dart';
+import '../../base/extensions/error_model_extensions.dart';
+import '../../base/extensions/string_extensions.dart';
+import '../../base/models/credentials_model.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../base/models/user_model.dart';
+import '../../lib_auth/models/auth_token_model.dart';
+import '../../lib_auth/services/auth_service.dart';
+import '../../lib_permissions/services/permissions_service.dart';
+import '../../lib_router/blocs/router_bloc.dart';
+import '../../lib_router/router.dart';
+
+part 'onboarding_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the OnboardingBloC.
+abstract class OnboardingBlocEvents {
+  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
+  void setEmail(String email);
+
+  @RxBlocEvent(type: RxBlocEventType.behaviour, seed: '')
+  void setPassword(String password);
+
+  void register();
+
+  void resumeOnboarding();
```

Comment:
Feel free to include documentation on these events.

Reply:
Done